src/main/java/com/mercateo/rest/hateoas/client/impl/ResponseImpl.java

```diff
@@ -20,6 +22,9 @@
 
     protected final T value;
 
+    @NonNull
+    private URI previousUri;
```

Comment:
could you rename this to "uri" or something like that? As i get it right, this is the uri where the actual response is from?